lib/ProductOpener/DataQualityFood.pm

```diff
@@ -2696,6 +2696,67 @@ sub check_food_groups ($product_ref) {
 	return;
 }
 
+=encoding utf8
+
+=head2 check_incompatible_tags( PRODUCT_REF )
+
+Checks if 2 incompatible tags are assigned to the product
+
+To include more tags to this check, 
+add the property "incompatible:en" 
+at the end of code block in the taxonomy
+
+Example:
+en:Non-fair trade, Not fair trade
+fr:Non issu du commerce équitable
+incompatible_with:en: en:fair-trade
+
+=cut
+
+sub check_incompatible_tags ($product_ref) {
+
+	# list of tags having 'incompatible_with' properties
+	my @tags_to_check = ("categories", "labels");
+
+	foreach my $tag_to_check (@tags_to_check) {
+		$log->debug("check_incompatible_tags: tag_to_check $tag_to_check") if $log->debug();
+
+		my $incompatible_with_hash = get_all_tags_having_property($product_ref, $tag_to_check, "incompatible_with:en");
+
+		foreach my $key (keys %{$incompatible_with_hash}) {
+			my $value = %{$incompatible_with_hash}{$key};
```

Comment:
I would have named: $key --> $tag and $value -> $incompatible_tags

It makes it easier to read.